source/common/http/http2/codec_impl.cc

```diff
@@ -802,10 +802,14 @@ Status ConnectionImpl::protocolErrorForTest() {
 }
 
 Status ConnectionImpl::onBeforeFrameReceived(const nghttp2_frame_hd* hd) {
-  ENVOY_CONN_LOG(trace, "about to recv frame type={}, flags={}, stream_id={}", connection_,
-                 static_cast<uint64_t>(hd->type), static_cast<uint64_t>(hd->flags), hd->stream_id);
+  ENVOY_CONN_LOG(trace, "about to recv frame type={}, flags={}, stream_id={}, length={}",
+                 connection_, static_cast<uint64_t>(hd->type), static_cast<uint64_t>(hd->flags),
+                 hd->stream_id, hd->length);
   current_stream_id_ = hd->stream_id;
-
+  StreamImpl* stream = getStream(hd->stream_id);
+  if (stream != nullptr && hd->type != METADATA_FRAME_TYPE) {
```

Comment:
Since we're exposing data bytes as well, we want to count the DATA framing here... since it's not available at the HCM level.

Reply:
This callback is called for all frame types so it should count DATA frames as well.

Reply:
I think kevin means that I didn't count the framing overhead when I count the body bytes in ConnectionManager. I think counting data bytes is trivial for h2, but i am not sure how to count it for h1.

Reply:
Added a pointer on where to count encoded body bytes for H1 more accurately. For decoding, I'd say the main spot we're missing is chunking.

We probably want to modify the onChunkHeader  to expose the content length as well as count the `\r\n` for both the number and the chunk data, and the number of digits.
https://github.com/envoyproxy/envoy/blob/4506199dc7d735e0d30e0c9e3fe4805e8da56977/source/common/http/http1/legacy_parser_impl.cc#L84

e.g. `4\r\nfooo\r\n` should count as 9 body bytes vs `fooo` which the decode will get (4 body bytes)